air/src/stack/mod.rs

```diff
@@ -1,3 +1,70 @@
-use super::EvaluationFrame;
+use super::{EvaluationFrame, Felt, FieldElement, TransitionConstraintDegree, Vec};
 
+mod field_ops;
 pub mod op_flags;
+mod stack_manipulation;
+mod system_ops;
+
+// STACK OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the stack module and all stack operation components.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    // stack manipulation operations contraints degrees.
+    let mut degrees = stack_manipulation::get_transition_constraint_degrees();
+    // system operations contraints degrees.
+    degrees.append(&mut system_ops::get_transition_constraint_degrees());
+    // field operations contraints degrees.
+    degrees.append(&mut field_ops::get_transition_constraint_degrees());
+
+    degrees
+}
+
+/// Returns the number of transition constraints for the stack operations.
+pub fn get_transition_constraint_count() -> usize {
+    stack_manipulation::get_transition_constraint_count()
+        + system_ops::get_transition_constraint_count()
+        + field_ops::get_transition_constraint_count()
+}
+
+/// Enforces constraints for the stack module and all stack operations.
+pub fn enforce_constraints<E: FieldElement<BaseField = Felt>>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+) -> usize {
+    let mut index = 0;
+
+    let frame_copy = frame.clone();
+
+    let op_flag = op_flags::OpFlags::new(frame_copy);
+
+    index += enforce_unique_constraints(frame, result, &op_flag);
+
+    index
+}
+
+/// Enforces unique constraints for all stack ops.
+pub fn enforce_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &op_flags::OpFlags<E>,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // stack manipulation operations transition constraints.
+    stack_manipulation::enforce_constraints(frame, result, op_flag);
+
+    constraint_offset += stack_manipulation::get_transition_constraint_count();
+
+    // system operations transition constraints.
+    system_ops::enforce_constraints(frame, &mut result[constraint_offset..], op_flag);
```

Comment:
minor nit: let's put system ops before stack manipulation ops (I think that's how we usually do it).

Reply:
I've kept the following ordering this time:
1) System ops
2) Field ops
3) Stack manipualtion ops